feature-libs/checkout/components/components/delivery-mode/delivery-mode.component.ts

```diff
@@ -73,26 +73,31 @@ export class DeliveryModeComponent implements OnInit, OnDestroy {
           this.checkoutDeliveryService
             .getSelectedDeliveryMode()
             .pipe(
-              map((deliveryMode: DeliveryMode) =>
-                deliveryMode && deliveryMode.code ? deliveryMode.code : null
+              map(
+                (deliveryMode: DeliveryMode | null | undefined) =>
+                  deliveryMode?.code
               )
             )
         )
       )
-      .subscribe(([deliveryModes, code]: [DeliveryMode[], string]) => {
-        if (
-          !(
-            code &&
-            !!deliveryModes.find((deliveryMode) => deliveryMode.code === code)
-          )
-        ) {
-          code = this.checkoutConfigService.getPreferredDeliveryMode(
-            deliveryModes
-          );
+      .subscribe(
+        ([deliveryModes, code]: [DeliveryMode[], string | undefined]) => {
+          if (
+            !(
+              code &&
+              !!deliveryModes.find((deliveryMode) => deliveryMode.code === code)
+            )
+          ) {
+            code = this.checkoutConfigService.getPreferredDeliveryMode(
+              deliveryModes
+            );
+          }
+          if (code) {
```

Comment:
Perhaps we can test if code !== undefined to prevent other false values from interfering

Reply:
code is `string | undefined`, it cannot be anything else.

Reply:
code is `string | undefined`, it cannot be anything else.